### PR TITLE
feat: add docs for C8 backup and restore

### PR DESCRIPTION
**Pull request by deepthidevaki**
## What is the purpose of the change

This PR add docs on how to backup C8 self-managed cluster.

depends on #1342 

related to #1297 

~NOTE:- Do not merge yet. The links in the docs needs to be updated after #1348, #1343 and #1344 are merged.~ These dependent PRs have been merged.

## Are there related marketing activities

_Are there any marketing activities related to the related feature or component? If so, please briefly describe them._

## When should this change go live?

_Does this change need to be released before a certain date or milestone? If so, when?_

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/self-managed/backup-restore/backup-and-restore.md
Comment (oleschoenburg) [inline docs/self-managed/backup-restore/backup-and-restore.md:45]: Might be worth mentioning here again that resuming zeebe exporting might be required too. That's the only step that has to be rolled back :disappointed:

File: docs/self-managed/backup-restore/backup-and-restore.md
Comment (oleschoenburg) [inline docs/self-managed/backup-restore/backup-and-restore.md:54]: Before merging we should link to the relevant sections here

**Review by oleschoenburg — APPROVED** [review]

File: sidebars.js
Comment (pepopowitz) [inline sidebars.js:982]: Due to the fact that we now have separate versions for Optimize than the rest of the docs, parallel sidebar changes will need to be made at https://github.com/camunda/camunda-platform-docs/blob/main/optimize_sidebars.js#L1540.

**Review by pepopowitz — DISMISSED** [review]
Additional changes are required to keep the sidebars from drifting in two separate areas of the site. 

See #1116 for more details on why these changes are needed, or let me know if I can further clarify!

File: docs/self-managed/backup-restore/backup-and-restore.md
Comment (christian-konrad) [inline docs/self-managed/backup-restore/backup-and-restore.md:7]: "Self-Managed" with an dash

File: docs/self-managed/backup-restore/zeebe-backup-and-restore.md
Comment (christian-konrad) [inline docs/self-managed/backup-restore/zeebe-backup-and-restore.md:9]: I propose to make this even more transparent. If it is experimental, people should not rely on it to make backups, imho.
For Web Modeler Self-Managed Beta, for instance, we use this disclaimers:

> It is not recommended for production use, and there is no maintenance service guaranteed. Special [terms & conditions](https://camunda.com/legal/terms/camunda-platform/camunda-platform-8-self-managed/) apply.

See
https://docs.google.com/document/d/1G3VOJfvtDxSf4kk5YM072XELZK6tBCtxkrop76tyl80/edit#heading=h.iqjs32b4jjoa

We'll introduce the terms "alpha" and "beta" here to avoid ambiguity:  https://github.com/camunda/camunda-platform-docs/pull/1304 

Linking to that docs page could be helpful then

Comment (deepthidevaki) [reply docs/self-managed/backup-restore/zeebe-backup-and-restore.md:9]: @christian-konrad I discussed this with @felix-mueller . We would like to offer this feature with production support. The only disclaimer is that API could change in the next release. We do not want to tag it as beta because then we will see no adoption and probably almost no feedback.  I can remove "experimental feature" from the disclaimer. Is that ok?

**Comment by deepthidevaki** [comment]
@akeller I have cross linked the new pages from the #1342 #1343, #1344, #1348 in this PR. So this PR would not build. Since we are short on time, I'm already requesting your review. You can also wait until other PRs are merged.

**Comment by akeller** [comment]
@deepthidevaki all other PRs have been merged. I will move forward with reviewing and merging!

File: docs/self-managed/backup-restore/backup-and-restore.md
Comment (akeller) [inline docs/self-managed/backup-restore/backup-and-restore.md:43]:
```suggestion
12. Resume exporting in Zeebe. (See [Zeebe management api](/self-managed/zeebe-deployment/operations/))
```

**Review by akeller — DISMISSED** [review]

File: docs/self-managed/backup-restore/backup-and-restore.md
Comment (akeller) [inline docs/self-managed/backup-restore/backup-and-restore.md:47]:
```suggestion
12. Resume exporting in Zeebe. (See [Zeebe management api](/self-managed/zeebe-deployment/operations/management-api.md))
```

**Review by akeller — APPROVED** [review]
Clearly my brain doesn't want to actually fix the link that will let this merge 😵

File: docs/self-managed/backup-restore/zeebe-backup-and-restore.md
Comment (christian-konrad) [reply docs/self-managed/backup-restore/zeebe-backup-and-restore.md]: Perfect!

**Pull request merged**
